tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFileOverride.java

```diff
@@ -0,0 +1,168 @@
+package org.unicode.cldr.util;
+
+import com.ibm.icu.util.VersionInfo;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import org.unicode.cldr.util.XPathParts.Comments;
+
+public class CLDRFileOverride extends CLDRFile {
+
+    /**
+     * Overrides the values of the sourceFile. The keys of the overrides must already be in the
+     * XMLSource source
+     */
+    public CLDRFileOverride(CLDRFile sourceFile, Map<String, String> valueOverrides) {
+        super(new XMLSourceMapOverride(sourceFile.dataSource, valueOverrides));
+    }
```

Comment:
I don't see the value for this being a CLDRFile subclass, since there's only a call to the constructor.

I think better would be to just make the XMLSourceMapOverride public (and the top level class) and add (if even needed) a convenience factory function:

```java
    public static final CLDRFile forFile(CLDRFile sourceFile, Map<String, String> valueOverrides) {
         return new CLDRFile(new XMLSourceMapOverride(sourceFile.dataSource, valueOverrides));
     }
```


but this isn't fundamentally a new kind of CLDRFile, so I don't think the subclass is warranted.